bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoCrockpotHandler.java

```diff
@@ -0,0 +1,307 @@
+        logger.debug("Received pair '{}':'{}' (service '{}') for thing '{}'", variable, value, service,
+                this.getThing().getUID());
+
+        updateStatus(ThingStatus.ONLINE);
+        this.stateMap.put(variable, value);
+    }
+
+    private synchronized void onSubscription() {
+        if (service.isRegistered(this)) {
+            logger.debug("Checking WeMo GENA subscription for '{}'", this);
+
+            String subscription = "basicevent1";
+
+            if ((subscriptionState.get(subscription) == null) || !subscriptionState.get(subscription).booleanValue()) {
+                logger.debug("Setting up GENA subscription {}: Subscribing to service {}...", getUDN(), subscription);
+                service.addSubscription(this, subscription, SUBSCRIPTION_DURATION);
+                subscriptionState.put(subscription, true);
+            }
+
+        } else {
+            logger.debug("Setting up WeMo GENA subscription for '{}' FAILED - service.isRegistered(this) is FALSE",
+                    this);
+        }
+    }
+
+    private synchronized void removeSubscription() {
+        logger.debug("Removing WeMo GENA subscription for '{}'", this);
+
+        if (service.isRegistered(this)) {
+            String subscription = "basicevent1";
+
+            if ((subscriptionState.get(subscription) != null) && subscriptionState.get(subscription).booleanValue()) {
+                logger.debug("WeMo {}: Unsubscribing from service {}...", getUDN(), subscription);
+                service.removeSubscription(this, subscription);
+            }
+
+            subscriptionState.remove(subscription);
+            service.unregisterParticipant(this);
+        }
+    }
+
+    private synchronized void onUpdate() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Configuration config = getThing().getConfiguration();
+            int refreshInterval = DEFAULT_REFRESH_INTERVAL_SECONDS;
+            Object refreshConfig = config.get("refresh");
+            refreshInterval = refreshConfig == null ? DEFAULT_REFRESH_INTERVAL_SECONDS
+                    : ((BigDecimal) refreshConfig).intValue();
+            refreshJob = scheduler.scheduleWithFixedDelay(refreshRunnable, 0, refreshInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    private boolean isUpnpDeviceRegistered() {
+        return service.isRegistered(this);
+    }
+
+    @Override
+    public String getUDN() {
+        return (String) this.getThing().getConfiguration().get(UDN);
+    }
+
+    /**
+     * The {@link updateWemoState} polls the actual state of a WeMo device and
+     * calls {@link onValueReceived} to update the statemap and channels..
+     *
+     */
+    protected void updateWemoState() {
+        String action = "GetCrockpotState";
+        String actionService = "basicevent";
+
+        String soapHeader = "\"urn:Belkin:service:" + actionService + ":1#" + action + "\"";
+        String content = "<?xml version=\"1.0\"?>"
```

Comment:
All these string seem static and could be made to static final fields.

Reply:
Could be, but it is the same in all handlers and I don't want to touch everything. Put it on my todo list for later updates.